### ISSUE TITLE
ImportError: cannot import name 'AsyncIOScheduler'

### ISSUE THREAD

**Issue by tachen-cs** [MainRo/deepspeech-server#21]
when I run the following sentense to config the deepspeech-server:
deepspeech-server --config config.json

An importError appears (I put the deepseech model downloaded ("model") just under the "deepspeech-server" folder ). Here is the bug information: 

Traceback (most recent call last):
  File "/usr/local/bin/deepspeech-server", line 4, in <module>
    __import__('pkg_resources').run_script('deepspeech-server==1.0.0', 'deepspeech-server')
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 658, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1445, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/local/lib/python3.6/dist-packages/deepspeech_server-1.0.0-py3.6.egg/EGG-INFO/scripts/deepspeech-server", line 3, in <module>
    __requires__ = 'deepspeech-server==1.0.0'
  File "/usr/local/lib/python3.6/dist-packages/deepspeech_server-1.0.0-py3.6.egg/deepspeech_server/server.py", line 4, in <module>
ImportError: cannot import name 'AsyncIOScheduler'

**Comment by tachen-cs** [comment]
After I reinstall wheels in the requirements.txt,  the bug information changed to:

Traceback (most recent call last):
  File "/usr/local/bin/deepspeech-server", line 4, in <module>
    __import__('pkg_resources').run_script('deepspeech-server==1.0.0', 'deepspeech-server')
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 658, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1445, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/local/lib/python3.6/dist-packages/deepspeech_server-1.0.0-py3.6.egg/EGG-INFO/scripts/deepspeech-server", line 3, in <module>
    __requires__ = 'deepspeech-server==1.0.0'
  File "/usr/local/lib/python3.6/dist-packages/deepspeech_server-1.0.0-py3.6.egg/deepspeech_server/server.py", line 16, in <module>
  File "/usr/local/lib/python3.6/dist-packages/deepspeech_server-1.0.0-py3.6.egg/deepspeech_server/deepspeech.py", line 9, in <module>
ImportError: cannot import name 'Model'

**Comment by MainRo** [comment]
did you install the deepspeech package before ? This dependency is not installed because you can choose to install the CPU or GPU implementation see the readme:

pip3 install deepspeech
or 
pip3 install deepspeech-gpu

**Comment by tachen-cs** [comment]
I found it might be the error brought by the deepspeech verison? ,first time I install deepspeech==0.1.1, and it appears previous bug information.

Then I install deepspeech package using "pip3 install deepspeech==0.4.1" , and the "python3 setup.py install“ run successfully,  then I download a pre-trained model (I need to use the model with v0.1) and put it under deepspeech-server folder, when I run  "deepspeech-server --config config.json"
another issue appears may be because of the feature parameters not suit (raise an error 5) .

So I am wondering how to config the deepspeech model with v0.1, I tried to write the config.json like this, but it doesn't work. I am not sure how to write the config.json for using default features from the model.

{
  "deepspeech": {
    "model" :"models/output_graph.pb",
    "alphabet": "models/alphabet.txt",
    "lm": "models/lm.binary",
    "trie": "models/trie",
    "features": "null"
  },
  "server": {
    "http": {
      "host": "0.0.0.0",
      "port": 8080,
      "request_max_size": 1048576
    }
  },
  "log": {
    "level": [
      { "logger": "deepspeech_server", "level": "DEBUG"}
    ]
  }
}

**Comment by taochencs** [comment]
Here is the error message:

initialize: Initialize(model='models/output_graph.pb', alphabet='models/alphabet.txt', lm='models/lm.binary', trie='models/trie', features=FeaturesParameters(n_features=26, n_context=9, beam_width=500, lm_weight=1.5, vwc_weight=2.25))
creating model models/output_graph.pb models/alphabet.txt with features FeaturesParameters(n_features=26, n_context=9, beam_width=500, lm_weight=1.5, vwc_weight=2.25)...
TensorFlow: v1.12.0-10-ge232881
DeepSpeech: v0.4.1-0-g0e40db6
Warning: reading entire model file into memory. Transform model file into an mmapped graph to reduce heap usage.
2019-05-28 23:57:02.695658: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
Not found: Op type not registered 'Assert' in binary running on gpu-server. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.
Exception in callback None()
handle: <Handle cancelled>
Traceback (most recent call last):
  File "/usr/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/concurrency/mainloopscheduler/asyncioscheduler.py", line 27, in interval
    disposable.disposable = self.invoke_action(action, state)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/concurrency/schedulerbase.py", line 14, in invoke_action
    ret = action(self, state)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/linq/observable/subscribeon.py", line 34, in action
    d.disposable = ScheduledDisposable(scheduler, source.subscribe(observer))
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observablebase.py", line 83, in subscribe
    current_thread_scheduler.schedule(set_disposable)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/concurrency/currentthreadscheduler.py", line 50, in schedule
    return self.schedule_relative(timedelta(0), action, state)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/concurrency/currentthreadscheduler.py", line 67, in schedule_relative
    Trampoline.run(queue)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/concurrency/currentthreadscheduler.py", line 31, in run
    item.invoke()
  File "/home/ct/.local/lib/python3.6/site-packages/rx/concurrency/scheduleditem.py", line 17, in invoke
    ret = self.scheduler.invoke_action(self.action, self.state)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/concurrency/schedulerbase.py", line 14, in invoke_action
    ret = action(self, state)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/linq/observable/returnvalue.py", line 30, in action
    observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/linq/observable/merge.py", line 128, in on_next
    observer.on_next(x)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/linq/observable/merge.py", line 128, in on_next
    observer.on_next(x)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/subjects/subject.py", line 92, in on_next
    observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/cyclotron_std/logging.py", line 56, in on_request_item
    observer.on_next(i)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/linq/observable/take.py", line 37, in on_next
    observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/linq/observable/combinelatest.py", line 91, in on_next
    next(i)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/linq/observable/combinelatest.py", line 72, in next
    observer.on_next(res)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/linq/observable/select.py", line 36, in on_next
    observer.on_next(result)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/linq/observable/merge.py", line 128, in on_next
    observer.on_next(x)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/subjects/subject.py", line 92, in on_next
    observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "/home/ct/.local/lib/python3.6/site-packages/rx/core/anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "/usr/local/lib/python3.6/dist-packages/deepspeech_server-1.0.0-py3.6.egg/deepspeech_server/deepspeech.py", line 90, in <lambda>
    sink.speech.subscribe(lambda item: on_deepspeech_request(item))
  File "/usr/local/lib/python3.6/dist-packages/deepspeech_server-1.0.0-py3.6.egg/deepspeech_server/deepspeech.py", line 84, in on_deepspeech_request
    item.model, item.alphabet, item.lm, item.trie, item.features or FeaturesParameters())
  File "/usr/local/lib/python3.6/dist-packages/deepspeech_server-1.0.0-py3.6.egg/deepspeech_server/deepspeech.py", line 48, in setup_model
    features.n_features, features.n_context, alphabet, features.beam_width)
  File "/home/ct/.local/lib/python3.6/site-packages/deepspeech/__init__.py", line 14, in __init__
    raise RuntimeError("CreateModel failed with error code {}".format(status))
RuntimeError: CreateModel failed with error code 5

**Comment by tachen-cs** [comment]
I test several version of deespeech in pypi. And found if I want to use a pre-trained model with v0.1. Then I have to use deepspeech==0.1.0~0.2.0a8. And I found if I use these versions of deepspeech in pypi, then the error becomes ImportError: cannot import name 'Model'

If I use a more advanced version of deepspeech, it may not suit for the model v0.1, it has the following errors:
RuntimeError: CreateModel failed with error code 5.

Looks like I have to modify your code to suit deepseech==0.1.0 in pypi, can you give me some hints?

Thank you very much!

**Comment by tachen-cs** [comment]
I successfully install deepspeech-server with model v0.1. Just simply modify deepspeech_server/deepspeech.py following code from https://github.com/mozilla/DeepSpeech/blob/v0.1.1/native_client/python/client.py

Thank you very much!

**Comment by MainRo** [comment]
yes the client API changed after 0.1. The code of deepspeech-server has been adapted at that time. You will also find the old code in the history of deepspeech-server.
glad you found a solution for your issue.
